From 2b1153a82615f4ce9985b7008b00fdc60b47c6a2 Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@ubuntu.com>
Date: Wed, 28 May 2025 11:17:53 +0200
Subject: [PATCH] Add copy of toOUString() from QtFrameGrabber.cxx
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

<<PKGBUILDDIR>>/avmedia/source/qt6/QtPlayer.cxx: In member function ‘virtual com::sun::star::uno::Reference<com::sun::star::media::XFrameGrabber> avmedia::qt::QtPlayer::createFrameGrabber()’:
/<<PKGBUILDDIR>>/avmedia/source/qt6/QtPlayer.cxx:295:42: error: ‘toOUString’ was not declared in this scope; did you mean ‘toQString’?
  295 |             = createPlatformFrameGrabber(toOUString(m_xMediaPlayer->source().url()));
      |                                          ^~~~~~~~~~
      |                                          toQString

Change-Id: Ie016a3138924722fd7c035ccb5faaf9fb71c7777
---
 avmedia/source/qt6/QtPlayer.cxx | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/avmedia/source/qt6/QtPlayer.cxx b/avmedia/source/qt6/QtPlayer.cxx
index dc0dea9fb05a..a8b1d27f0131 100644
--- a/avmedia/source/qt6/QtPlayer.cxx
+++ b/avmedia/source/qt6/QtPlayer.cxx
@@ -41,6 +41,14 @@
 
 using namespace ::com::sun::star;
 
+namespace
+{
+inline OUString toOUString(const QString& s)
+{
+    return OUString(reinterpret_cast<const sal_Unicode*>(s.data()), s.length());
+}
+}
+
 namespace avmedia::qt
 {
 QtPlayer::QtPlayer()
-- 
2.48.1

